### PR TITLE
Adding rainbow module.

### PR DESCRIPTION
**Pull request by kattni**
Adds rainbow module with `colorwheel()` function. Removes `colorwheel()` from `_pixelbuf`.

### CODE REVIEW

File: shared-bindings/_pixelbuf/__init__.c [hunk 1]
```diff
@@ -41,40 +41,9 @@
...
-    } else {
-        pos -= 170;
-        return (uint8_t)(pos * 3) << 16 | (uint8_t)(255 - (pos * 3));
-    }
-}
-
 STATIC const mp_rom_map_elem_t pixelbuf_module_globals_table[] = {
     { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR__pixelbuf) },
     { MP_ROM_QSTR(MP_QSTR_PixelBuf), MP_ROM_PTR(&pixelbuf_pixelbuf_type) },
-    { MP_ROM_QSTR(MP_QSTR_wheel), MP_ROM_PTR(&pixelbuf_colorwheel_obj) },
```
Comment (gamblor21) [inline shared-bindings/_pixelbuf/__init__.c:76]: I am assuming you probably did not want this keyword to continue on? I did not see it defined in the new module.

**Review by gamblor21 — COMMENTED** [review]
Did a quick look, everything seems fine with me. One question but probably something you did on purpose.

File: shared-bindings/_pixelbuf/__init__.c (hunk 1 above)
Comment (kattni) [reply shared-bindings/_pixelbuf/__init__.c:76]: You are correct! I did not want it to continue on.

**Comment by dhalbert** [comment]
The one failure is that some `escornabot_makech` builds were too big. That board is a robot with wheels, etc. Based on that, you could turn off MIDI and I think that would not crimp any uses.

In `ports/atmel_samd/boards/escornabot_makech/mpconfigboard.mk`, add `CIRCUITPY_USB_MIDI = 0`. An example of that is in `adafruit_proxlight_trinkey_m0/mpconfigboard.mk` (along with other things).

Once #4903 is merged, we won't be so close to the edge on many builds.

**Comment by kattni** [comment]
@dhalbert Fix submitted. Also updated copyright in `_pixelbuf` that I meant to include in the first place, but did not want to restart the entire CI process for only that change.

**Review by dhalbert — APPROVED** [review]
Thanks and congrats!

**Comment by dhalbert** [comment]
Partly addresses #4976.

**Comment by tannewt** [comment]
Sorry I just looked at this. We shouldn't call this `rainbow` because of the existing pypi package: https://pypi.org/project/rainbow/

We could call it `adafruit_rainbow` though. Or `rainbowio` :-)

**Comment by dhalbert** [comment]
`adafruit_rainbow` is ok by me. or `pixel_colors`, or `rgb`, or something like that.

**Comment by dhalbert** [comment]
`adafruit_rainbow` is ok by me, or even `rainbowio`. Or:
- `rainbows`  (not taken in PyPi)
 - `pixel_colors`
- `rgb`
- `color_math`
- `color_calc`

**Comment by kattni** [comment]
So I really like `rainbowio` but on the off chance we have to make a Python library version of `colorwheel()`, I should perhaps accept `adafruit_rainbow` so the name would be the same and all the code would work. It comes down to, what is the likelihood that we'll have to do a Python version? The only reason the Python version of Pixelbuf exists is that `_pixelbuf` was too big for tiny builds. `rainbow` built fine into all but one build, so I'm inclined to think we're not going to need a Python version. I'd like to avoid needing a Python version, but if that's going to happen, we'll end up in a `_pixelbuf`/`adafruit_pypixelbuf` situation again with `rainbowio`.

**Comment by tannewt** [comment]
We could make a `rainbowio` pypi package if we really wanted to.

**Pull request merged**